File: app/src/main/java/com/chris/eban/data/RoomTypeConverters.java
package com.chris.eban.data;

import java.util.Date;

import androidx.room.TypeConverter;

public class RoomTypeConverters {

    @TypeConverter
    public static Date fromTimeStamp(Long timeStamp) {
        return timeStamp == null ? null : new Date(timeStamp);
    }

    public static Long dateToTimeStamp(Date date) {
        return date == null ? null : date.getTime();
    }
}

File: app/src/main/java/com/chris/eban/data/RoomTypeConverters.kt
package com.chris.eban.data

import androidx.room.TypeConverter
import java.util.*


class RoomTypeConverters {


    @TypeConverter
    fun fromTimeStamp(timeStamp: Long?): Date? {
        return if (timeStamp == null) null else Date(timeStamp)
    }


    @TypeConverter
    fun dateToTimeStamp(date: Date): Long = date.time

    fun dateToTimeStamp(date: Date?): Long? {
        return date?.time
    }

}
